test/conftest.py

```diff
@@ -940,7 +934,8 @@ def adaptive_model_qa(num_processes):
     # check if all workers (sub processes) are closed
     current_process = psutil.Process()
     children = current_process.children()
-    assert len(children) == 0
+    if len(children) != 0:
+        logging.error(f"Not all the subprocesses are closed! {len(children)} are still running.")
```

Comment:
I wonder what does this entail? If the situation is not recoverable, killing the process with an assert might well be a solution...

Reply:
I do not fully understand why we had that assert in the first place: I believe we had issues of leaking child processes what would kill the CI. And with the workflow change, somehow this check started failing.

However, now that I replaced the assert with a log line, I see that it consistently reports 8 leftover children, and the number does not grow. Somehow it just takes a few milliseconds more to terminate those children, but they eventually get terminated.

Do you think it's too dangerous to leave it like this? I considered adding a small sleep here, but I believe sleep in tests are the worst thing possible :smile: So if I can avoid them I will. Let me know :slightly_smiling_face:

Reply:
The fixture is called 3 times: with 0, 2 and MAXPROC-1 workers, where does 8 come from? Plus the library we call at some point calls `self.process_pool.join()` so I'm getting suspicious our `children = current_process.children()` does what we think it does... For the scope of this PR, given our doubts I think your change is the best course of action.

Reply:
I now remember I even opened an issue about this, when I discussed this change with Julian on the old PR: https://github.com/deepset-ai/haystack/issues/2539  I guess that's enough for now